FreeRTOS/Test/CMock/event_groups/event_groups_utest.c

```diff
@@ -0,0 +1,608 @@
+    xTaskGetSchedulerState_IgnoreAndReturn( taskSCHEDULER_SUSPENDED );
+    vTaskPlaceOnUnorderedEventList_Ignore();
+    uxTaskResetEventItemValue_IgnoreAndReturn( 0 );
+
+    /* Set-up */
+    EventBits_t uxReturn;
+    StaticEventGroup_t xCreatedEventGroup = { 0 };
+    xEventGroupHandle = xEventGroupCreateStatic( &xCreatedEventGroup );
+
+    /* API to Test */
+    /* Test-1: Set BIT_0 before reach the sync point and wait for ALL_SYNC_BITS is set */
+    uxReturn = xEventGroupSync( xEventGroupHandle, BIT_0, ALL_SYNC_BITS, portMAX_DELAY );
+
+    /* Validate: Expect to timeout and BIT_0 was set */
+    TEST_ASSERT_EQUAL( BIT_0, uxReturn );
+
+    /* Test-2: BIT_2 was set, no block */
+    uxReturn = xEventGroupSetBits( xEventGroupHandle, BIT_2 );
+    uxReturn = xEventGroupSync( xEventGroupHandle, 0, ALL_SYNC_BITS, 0 );
+
+    /* Validate: Expect to return and BIT_0 and BIT_2 was set */
+    TEST_ASSERT_EQUAL( ( BIT_0 | BIT_2 ), uxReturn );
+
+    /* Test-2: BIT_4 was set, now ALL_SYNC_BITS are set */
+    uxReturn = xEventGroupSetBits( xEventGroupHandle, BIT_4 );
+    uxReturn = xEventGroupSync( xEventGroupHandle, 0, ALL_SYNC_BITS, portMAX_DELAY );
+
+    /* Validate: Expect to return and BIT_0 and BIT_2 was set */
+    TEST_ASSERT_EQUAL( ( BIT_0 | BIT_2 | BIT_4 ), uxReturn );
+}
+
+void test_uxEventGroupGetNumber_Success( void )
+{
+    /* Expectation of Function: xEventGroupCreate */
+    vListInitialise_Expect( 0 );
+    vListInitialise_IgnoreArg_pxList();
+    vListInitialise_ReturnThruPtr_pxList( pxListTemp );
+
+    /* Set-up */
+    UBaseType_t xReturn;
+    StaticEventGroup_t xCreatedEventGroup = { 0 };
+    xEventGroupHandle = xEventGroupCreateStatic( &xCreatedEventGroup );
+
+    /* API to Test */
+    xReturn = uxEventGroupGetNumber( NULL );
+    TEST_ASSERT_EQUAL( 0, xReturn );
+    xReturn = uxEventGroupGetNumber( xEventGroupHandle );
+    TEST_ASSERT_EQUAL( 0, xReturn );
+}
+
+void test_vEventGroupSetNumber_Success( void )
+{
+    /* Expectation of Function: xEventGroupCreate */
+    vListInitialise_Expect( 0 );
+    vListInitialise_IgnoreArg_pxList();
+    vListInitialise_ReturnThruPtr_pxList( pxListTemp );
+
+    /* Set-up */
+    UBaseType_t xReturn;
+    StaticEventGroup_t xCreatedEventGroup = { 0 };
+    xEventGroupHandle = xEventGroupCreateStatic( &xCreatedEventGroup );
+
+    /* API to Test */
+    vEventGroupSetNumber( xEventGroupHandle, 3 );
+    xReturn = uxEventGroupGetNumber( xEventGroupHandle );
+
+    /* Validate */
+    TEST_ASSERT_EQUAL( 3, xReturn );
+}
+
+
+/* --- Below testcase are meaningless, currently just added for code coverage --- */
+
+
+void test_xEventGroupClearBitsFromISR_Success( void )
+{
+    /* Expectation of Function: xEventGroupClearBitsFromISR */
+    xTimerPendFunctionCallFromISR_IgnoreAndReturn( pdPASS );
+
+    /* API to Test */
+    ( void ) xEventGroupClearBitsFromISR( NULL, BIT_0 );
+}
+
+void test_xEventGroupSetBitsFromISR_Success( void )
```

Comment:
You need to verify the outcome of this call and that the bits are actually set.
Also missing a coverage tag.

Reply:
This is the function that hard to write a test case for it. The resigter the vEventGroupSetBitsCallback to the timer module. The callback function just calls the xEventGroupSetBits. Both vEventGroupSetBitsCallback and xEventGroupSetBits are covered by the test, so I think addind a test for this function is kind of meaningless.

Reply:
Similar to my suggestion for xEventGroupClearBitsFromISR, You basically need to verify that the function isn't doing much of anything and is just passing parameters into the xTimerPendFunctionCallFromISR mock and returning the mocked return value.